sdk/src/test/java/com/hedera/hashgraph/sdk/ClientTest.java

```diff
@@ -134,21 +138,101 @@ void testFromJson() throws Exception {
+        Assertions.assertEquals(network.get("0.testnet.hedera.com:50211"), new AccountId(3));
+        Assertions.assertEquals(network.get("1.testnet.hedera.com:50211"), new AccountId(4));
+
+        nodes.put("2.testnet.hedera.com:50211", new AccountId(5));
+
+        client.setNetwork(nodes);
+        network = client.getNetwork();
+
+        Assertions.assertEquals(3, network.size());
+        Assertions.assertEquals(network.get("0.testnet.hedera.com:50211"), new AccountId(3));
+        Assertions.assertEquals(network.get("1.testnet.hedera.com:50211"), new AccountId(4));
+        Assertions.assertEquals(network.get("2.testnet.hedera.com:50211"), new AccountId(5));
+
+        nodes.clear();
+        nodes.put("2.testnet.hedera.com:50211", new AccountId(5));
+
+        client.setNetwork(nodes);
+        network = client.getNetwork();
+
+        Assertions.assertEquals(1, network.size());
+        Assertions.assertEquals(network.get("2.testnet.hedera.com:50211"), new AccountId(5));
+
+        nodes.clear();
+        nodes.put("2.testnet.hedera.com:50211", new AccountId(6));
+
+        client.setNetwork(nodes);
+        network = client.getNetwork();
+
+        Assertions.assertEquals(1, network.size());
+        Assertions.assertEquals(network.get("2.testnet.hedera.com:50211"), new AccountId(6));
+
+        client.close();
+    }
+
+    @Test
+    @DisplayName("setNetwork() functions correctly")
+    void setMirrorNetworkWorks() throws Exception {
+        var nodes = new ArrayList<String>();
+        nodes.add("hcs.testnet.mirrornode.hedera.com:5600");
+
+        Client client = Client.forNetwork(new HashMap<>()).setMirrorNetwork(nodes);
+        @Var var mirrorNetwork = new HashSet<>(client.getMirrorNetwork());
+
+        Assertions.assertEquals(1, mirrorNetwork.size());
+        Assertions.assertTrue(mirrorNetwork.contains("hcs.testnet.mirrornode.hedera.com:5600"));
+
+        client.setMirrorNetwork(nodes);
+        mirrorNetwork = new HashSet<>(client.getMirrorNetwork());
+
+        Assertions.assertEquals(1, mirrorNetwork.size());
+        Assertions.assertTrue(mirrorNetwork.contains("hcs.testnet.mirrornode.hedera.com:5600"));
+
+        nodes.add("hcs.testnet1.mirrornode.hedera.com:5600");
+
+        client.setMirrorNetwork(nodes);
+        mirrorNetwork = new HashSet<>(client.getMirrorNetwork());
+
+        Assertions.assertEquals(2, mirrorNetwork.size());
+        Assertions.assertTrue(mirrorNetwork.contains("hcs.testnet.mirrornode.hedera.com:5600"));
+        Assertions.assertTrue(mirrorNetwork.contains("hcs.testnet1.mirrornode.hedera.com:5600"));
+
+        nodes.clear();
+        nodes.add("hcs.testnet1.mirrornode.hedera.com:5600");
+
+        client.setMirrorNetwork(nodes);
+        mirrorNetwork = new HashSet<>(client.getMirrorNetwork());
+
+        Assertions.assertEquals(1, mirrorNetwork.size());
+        Assertions.assertTrue(mirrorNetwork.contains("hcs.testnet1.mirrornode.hedera.com:5600"));
```

Comment:
```suggestion
        assertThat(client.getMirrorNetwork()).isEqualTo(deleteNode)
```

Reply:
A couple things to mention here:
 - The version of JUnit in the SDK does not have `assertThat`; we are on the latest version, but I'm guessing there are different JUnit libraries
 - `isEqualTo` would only work when `deleteNode` is exactly 1 element since `client.getMirrorNetwork()` is not guaranteed to return you a list in the same order

Reply:
The `assertThat()` that I'm recommending to use is from assert4j, which I added as a dependency in one of my PRs.

I'm aware about `isEqualTo()` only working if it's the same size and order. If you don't want to guarantee a stable order of `getMirrorNetwork()` you can use `assertThat(client.getMirrorNetwork()).containsExactlyInAnyOrder(item2, item1, item3)`.